packages/code/src/Code.tsx

```diff
@@ -169,16 +291,65 @@ function Code({
     </Syntax>
   );
 
+  function handleScroll(e: React.UIEvent) {
+    const { scrollWidth, clientWidth: elementWidth } = e.target as HTMLElement;
+    const isScrollable = scrollWidth > elementWidth;
+
+    if (isScrollable) {
+      const scrollPosition = (e.target as HTMLElement).scrollLeft;
+      const maxPosition = scrollWidth - elementWidth;
+
+      if (scrollPosition > 0 && scrollPosition < maxPosition) {
+        setScrollState(ScrollState.Both);
+      } else if (scrollPosition > 0) {
+        setScrollState(ScrollState.Left);
+      } else if (scrollPosition < maxPosition) {
+        setScrollState(ScrollState.Right);
+      }
+    }
+  }
+
+  const debounceScroll = debounce(handleScroll, 50, { leading: true });
+
   if (!multiline) {
     return (
       <div className={wrapperStyle}>
         {renderedWindowChrome}
 
         <div
-          {...(rest as DetailedElementProps<HTMLDivElement>)}
-          className={wrapperClassName}
+          className={css`
+            display: flex;
+          `}
         >
-          {renderedSyntaxComponent}
+          <div
+            {...(rest as DetailedElementProps<HTMLDivElement>)}
+            className={wrapperClassName}
+            onScroll={e => {
+              e.persist();
+              debounceScroll(e);
+            }}
+            ref={scrollableSingleLine}
+          >
+            {renderedSyntaxComponent}
+          </div>
+          {!showWindowChrome && (
+            <div className={cx(copyStyle, getSidebarVariantStyle(variant))}>
+              <CopyToClipboard
+                text={content}
+                onCopy={() => {
+                  setCopied(true);
```

Comment:
We should unset this state after about a second so that the checkmark doesn't persist forever.